shared/util/saga.js

```diff
@@ -0,0 +1,11 @@
+// @flow
+
+import {put, cps} from 'redux-saga/effects'
+
+export function * cpsWithWaiting (waitingFn: (isWaiting: boolean) => any, cpsFn: any, ...args: any): any {
+  yield put(waitingFn(true))
```

Comment:
we could do that to, but I think it's better if waitingFn is a normal function that returns an action.

Normal functions are going to be easier to reason about than generators, and they're slightly smaller.


also a slight tweak to your waiting definition:
```
function * waiting (waiting: boolean): SagaGenerator<any, any> {
   yield put({type: Constants.waiting, payload: {waiting}})
 }
```